### PR TITLE
Starting decade show view

### PR DESCRIPTION
**Pull request by C4L86**

### CODE REVIEW

File: app/models/release.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ class Release < ActiveRecord::Base
   has_many   :formats
 
   # Finds releases associated with a given decade and returns a hash with decade keys and Release Object values
+  def self.current_decade
+    Time.now.year - (Time.now.year % 10)
```
Comment (sumeetjain) [inline app/models/release.rb]: Calls time.now 2 times (DuplicateMethodCall)

Comment (sumeetjain) [inline app/models/release.rb]: Calls time.now.year 2 times (DuplicateMethodCall)

File: app/models/release.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ class Release < ActiveRecord::Base
   has_many   :formats
 
   # Finds releases associated with a given decade and returns a hash with decade keys and Release Object values
+  def self.current_decade
+    Time.current.year - (Time.current.year % 10)
```
Comment (sumeetjain) [inline app/models/release.rb:7]: Calls time.current 2 times (DuplicateMethodCall)

Comment (sumeetjain) [inline app/models/release.rb:7]: Calls time.current.year 2 times (DuplicateMethodCall)

File: app/models/release.rb [hunk 3]
```diff
@@ -2,6 +2,11 @@ class Release < ActiveRecord::Base
   belongs_to :album
   has_many   :formats
 
+  # Returns the current decade
+  def self.current_decade
+    Time.current.year - (Time.current.year % 10)
```
Comment (sumeetjain) [inline app/models/release.rb:7]: Calls time.current 2 times (DuplicateMethodCall)

Comment (sumeetjain) [inline app/models/release.rb:7]: Calls time.current.year 2 times (DuplicateMethodCall)

File: app/models/release.rb [hunk 4]
```diff
@@ -2,6 +2,11 @@ class Release < ActiveRecord::Base
   belongs_to :album
   has_many   :formats
 
+  # Returns the current decade
+  def self.current_decade
```
Comment (sumeetjain) [inline app/models/release.rb:6]: Remove unused methods (release#current_decade)

File: config/routes.rb [hunk 5]
```diff
@@ -42,5 +42,5 @@
   get "formats/:id"   => 'formats#show'
 
   get "decades/index" => 'decades#index'
-  get "decades/:id"   => 'decades#show'
+  get "decades/"   => 'decades#show'
```
Comment (sumeetjain) [inline config/routes.rb]: Operator `=>` should be surrounded by a single space.

**Comment by C4L86** [comment]
I'm attempting to define the show views for each decade but am unsure how to route to a given decade based on the albums that fall within. The only thing that I can think of would be to draw the decade key from the 'info_by_decade' method I have defined in the releases model. Could I come chat with you about this?

**Comment by halfghaninne** [comment]
If you'd like, I can take a look with you now! I'm upstairs.

File: app/controllers/decades_controller.rb [hunk 6]
```diff
@@ -8,6 +8,9 @@ def index
   end
 
   def show
+    @decade   = params[:year]
```
Comment (sumeetjain) [inline app/controllers/decades_controller.rb]: Calls params[:year] 2 times (DuplicateMethodCall)

File: app/controllers/decades_controller.rb [hunk 7]
```diff
@@ -8,6 +8,9 @@ def index
...
+    @decade_releases = (Release.info_by_decade)[params[:year].to_i]
```
Comment (sumeetjain) [inline app/controllers/decades_controller.rb]: Don't use parentheses around a method call.

**Pull request merged**